api/v1/verticadb_webhook.go

```diff
@@ -1701,6 +1728,202 @@ func (v *VerticaDB) checkSandboxPrimary(allErrs field.ErrorList, oldObj *Vertica
 	return allErrs
 }
 
+// checkNewSBoxOrSClusterShutdownUnset ensures a new subcluster or sandbox to be added to a vdb has Shutdown field set to false
+func (v *VerticaDB) checkNewSBoxOrSClusterShutdownUnset(allErrs field.ErrorList) field.ErrorList {
+	statusSClusterMap := v.GenStatusSubclusterMap()
+	statusSBoxMap := v.GenStatusSandboxMap()
+	for i := range v.Spec.Sandboxes {
+		newSBox := v.Spec.Sandboxes[i]
+		if _, foundInStatus := statusSBoxMap[newSBox.Name]; !foundInStatus && newSBox.Shutdown {
+			p := field.NewPath("spec").Child("sandboxes").Index(i).Child("shutdown")
+			err := field.Invalid(p,
+				newSBox.Shutdown,
+				fmt.Sprintf("shutdown must be false when adding sandbox %q",
+					newSBox.Name))
+			allErrs = append(allErrs, err)
+		}
+	}
+	for i := range v.Spec.Subclusters {
+		newSCluster := &v.Spec.Subclusters[i]
+		if _, foundInStatus := statusSClusterMap[newSCluster.Name]; !foundInStatus && newSCluster.Shutdown {
+			p := field.NewPath("spec").Child("subclusters").Index(i).Child("shutdown")
+			err := field.Invalid(p,
+				newSCluster.Shutdown,
+				fmt.Sprintf("shutdown must be false when adding subcluster %q",
+					newSCluster.Name))
+			allErrs = append(allErrs, err)
+		}
+	}
+	return allErrs
+}
+
+// findSclustersToUnsandboxreturn a map whose key is the pointer of a subcluster that is to be unsandboxed.
+// The value is the pointer to a sandbox where the subcluster lives in
+func (v *VerticaDB) findSclustersToUnsandbox(old runtime.Object) map[*Subcluster]*Sandbox {
+	oldObj := old.(*VerticaDB)
+	oldSubclusterInSandbox := oldObj.GenSubclusterSandboxMap()
+	newSuclusterInSandbox := v.GenSubclusterSandboxMap()
+	oldSubclusterMap := oldObj.GenSubclusterMap()
+	oldSandboxMap := oldObj.GenSandboxMap()
+	newSubclusterMap := v.GenSubclusterMap()
+	sclusterSboxMap := make(map[*Subcluster]*Sandbox)
+	for oldSubclusterName, oldSandboxName := range oldSubclusterInSandbox {
+		_, oldSubclusterInNewSandboxes := newSuclusterInSandbox[oldSubclusterName]
```

Comment:
You need to check if the old sandbox name and new sandbox name are the same. If we move a subcluster from a sandbox to another in new vdb, that subcluster is still in new vdb, but it will be unsandboxed.

Reply:
I will make code change to cover that scenario and set up a new test case.